File: ITapply/ITapply.Models/Requests/WorkExperienceInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.Requests
{
    public class WorkExperienceInsertRequest
    {
        [Required(ErrorMessage = "Candidate ID is required.")]
        [Range(1, int.MaxValue, ErrorMessage = "Candidate ID must be a positive number.")]
        public int CandidateId { get; set; }

        [Required(ErrorMessage = "Company name is required.")]
        [StringLength(200, ErrorMessage = "Company name cannot exceed 200 characters.")]
        public string CompanyName { get; set; }

        [Required(ErrorMessage = "Position is required.")]
        [StringLength(200, ErrorMessage = "Position cannot exceed 200 characters.")]
        public string Position { get; set; }

        [Required(ErrorMessage = "Start date is required.")]
        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        [StringLength(4000, ErrorMessage = "Description cannot exceed 4000 characters.")]
        public string? Description { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_desktop/lib/models/requests/work_experience_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'work_experience_insert_request.g.dart';

@JsonSerializable()
class WorkExperienceInsertRequest {
  final int candidateId;
  final String companyName;
  final String position;
  final DateTime startDate;
  final DateTime? endDate;
  final String? description;

  WorkExperienceInsertRequest({
    required this.candidateId,
    required this.companyName,
    required this.position,
    required this.startDate,
    this.endDate,
    this.description,
  });
  
  factory WorkExperienceInsertRequest.fromJson(Map<String, dynamic> json) => _$WorkExperienceInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$WorkExperienceInsertRequestToJson(this);
}